app/menus/projects/menu.rb

```diff
@@ -104,7 +104,7 @@ def public_filters
 
     def my_filters
       persisted_filters
-        .select { |query| query.user == current_user }
+        .select { |query| query.user == current_user && !query.public? }
```

Comment:
Maybe better:
```suggestion
        .reject(&:public?)
        .select { |query| query.user == current_user }
```
And same for `shared_filters`.

It is probably not yet worth somehow combining filtering for three methods to not repeat calls.

Reply:
I do not agree with this one. With `select` 3 times we are building and throwing away 3 arrays of AR objects. The checks are small enough and easy to understand what is filtered here, so I don't see a benefit of splitting those up.